docxcompose/composer.py

```diff
@@ -624,3 +647,42 @@ def fix_section_types(self, doc):
         first_new_section_idx = len(self.doc.sections) - len(doc.sections)
         self.doc.sections[first_new_section_idx].start_type = self.doc.sections[-1].start_type
         self.doc.sections[-1].start_type = doc.sections[-1].start_type
+
+    def fix_header_and_footers(self, doc):
+        """
+        The master document usually only has one section, hence its section
+        properties are defined directly in the body of the document and apply
+        to the last section of the document. For all other sections but the
+        last one, section properties are defined in the last paragraph of
+        the section.
+        Headers and footers are inherited from the previous section properties
+        if they are not defined in a given section. If not defined in the first
+        section, then blank headers and footers will be used., so we need to
+        make sure to add the definition from the main template in the first
+        section of the document if there are more than one sections.
+        """
+
+        if self.first_section_properties_added:
+            return
+
+        if len(self.doc.sections) == 1 or len(doc.sections) == 1:
+            return
+
+        first_new_section_idx = len(self.doc.sections) - len(doc.sections)
```

Comment:
is this safe? Isn't it possible to get a negative number here?

Reply:
I think so, it's already used exactly like this on line 647 above